samples/nrf9160/serial_lte_modem/src/slm_at_host.c

```diff
@@ -285,53 +345,52 @@ static void uart_rx_handler(u8_t character)
 
 	return;
 send:
-	uart_irq_rx_disable(uart_dev);
+	uart_rx_disable(uart_dev);
 	k_work_submit(&cmd_send_work);
 	at_buf_len = cmd_len;
 	cmd_len = 0;
 }
 
-static void isr(struct device *dev)
-{
-	u8_t character;
-
-	uart_irq_update(dev);
-
-	if (!uart_irq_rx_ready(dev)) {
-		return;
-	}
-
-	while (uart_fifo_read(dev, &character, 1)) {
-		uart_rx_handler(character);
-	}
-}
-
-static int at_uart_init(char *uart_dev_name)
+static void uart_callback(struct uart_event *evt, void *user_data)
 {
 	int err;
-	u8_t dummy;
+	char ch;
 
-	uart_dev = device_get_binding(uart_dev_name);
-	if (uart_dev == NULL) {
-		LOG_ERR("Cannot bind %s\n", uart_dev_name);
-		return -EINVAL;
-	}
-	/* Wait for the UART line to become valid */
-	do {
-		err = uart_err_check(uart_dev);
-		if (err) {
-			LOG_ERR("UART check failed: %d. "
-				"Dropping buffer and retrying.", err);
+	ARG_UNUSED(user_data);
 
-			while (uart_fifo_read(uart_dev, &dummy, 1)) {
-				/* Do nothing with the data */
-			}
-			k_sleep(10);
+	switch (evt->type) {
+	case UART_TX_DONE:
+		k_free(uart_tx_buf);
+		k_sem_give(&tx_done);
+		break;
+	case UART_TX_ABORTED:
+		k_free(uart_tx_buf);
+		k_sem_give(&tx_done);
+		LOG_INF("TX_ABORTED");
+		break;
+	case UART_RX_RDY:
+		ch = (char)*(evt->data.rx.buf + evt->data.rx.offset);
```

Comment:
Since only one byte buffers are used, this could be actually just `evt->data.rx.buf[0]`. This typecasting to `char` is also a bit weird, as `uart_rx_handler` takes a parameter of type `u8_t`.

Reply:
Updated, thanks